src/main/java/seedu/duke/command/ViewCommand.java

```diff
@@ -0,0 +1,88 @@
+package seedu.duke.command;
+
+import seedu.duke.data.UserData;
+import seedu.duke.event.Event;
+import seedu.duke.event.EventList;
+import seedu.duke.exception.DukeException;
+import seedu.duke.exception.InvalidIndexException;
+import seedu.duke.exception.InvalidListException;
+import seedu.duke.exception.WrongNumberFormatException;
+import seedu.duke.exception.WrongNumberOfArgumentsException;
+import seedu.duke.storage.Storage;
+import seedu.duke.ui.Ui;
+
+import java.util.ArrayList;
+
+/**
+ * Command to view notes.
+ */
+public class ViewCommand extends Command {
+    private int index;
+    private String event;
+
+    /**
+     * Constructor for viewing.
+     *
+     * @param command user's command.
+     */
+    public ViewCommand(String command) {
+        this.isExit = false;
+        this.command = command;
+    }
+
+    /**
+     * View notes.
+     *
+     * @param data    object of UserData class containing user's data.
+     * @param ui      containing the responses to print.
+     * @param storage with the save file path to write to.
+     * @throws DukeException catch any exception under Duke.
+     */
+    @Override
+    public void execute(UserData data, Ui ui, Storage storage) throws DukeException {
+        try {
+            parseUserCommand(command);
+            EventList list = data.getEventList(event);
+            Event eventRequested = list.getEventByIndex(index - 1);
+            if (eventRequested != null) {
+                ArrayList<String> existingNotes = eventRequested.getNotes();
+                ui.printViewNote(existingNotes);
+            } else {
+                throw new InvalidListException("No such event type");
+            }
+        } catch (InvalidIndexException e) {
+            throw new InvalidIndexException("Error, no such index is available!");
```

Comment:
Should these 2 exceptions be swapped

Reply:
there are three kinds of error that can occur here it seems. The first kind occurs if the user type in the wrong type of event. The second is that the index is incorrect. The third is if the index and type is valid, but there is no notes to display for that particular event. ... perhaps the two exceptions need to be swapped

Reply:
It shouldn't be swapped